### PR TITLE
News Topic Tabs

### PR DESCRIPTION
**Pull request by SirSaltyy**
Btw, since calling you Stephen I took all the code I added out of the HomeDashboard function and created a new function on its own page so I could export it and use it on the signed-out home page.

Still need to do:
-Curate content
-Make the rank and streak buttons inline with the header on the signed-in homepage
-Remove Topic Title Header on mobile

-Optional: Make the tabs CSS look better on mobile
-Controversial: Remove slideshow on signed-out mobile

### CODE REVIEW

File: web/components/news-topics-tabs.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+type Topic = {
+  title: string
+  content: ReactNode
+  onClick?: (event?: any) => void
+}
+
+type NewsTopicsTabsProps = { homeContent: ReactNode }
+
+export function TopicsTab({
```
Comment (mantikoros) [inline web/components/news-topics-tabs.tsx:22]: This probably doesn't merit being its own component. You're only using it in one place and there's minimal logic here.

File: web/components/news-topics-tabs.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  RedditBlackoutData,
+} from 'web/public/news-topics-data/newsTopicsData'
+
+type Topic = {
+  title: string
+  content: ReactNode
+  onClick?: (event?: any) => void
+}
+
+type NewsTopicsTabsProps = { homeContent: ReactNode }
```
Comment (mantikoros) [inline web/components/news-topics-tabs.tsx:20]: Don't define prop types outside the component if they're not being used anywhere else or are small.

File: web/components/news-topics-tabs.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      }))}
+    />
+  )
+}
+
+export function NewsTopicsTabs({ homeContent }: NewsTopicsTabsProps) {
+  const [selectedTab, setSelectedTab] = useState<string>('Home') // This selectedTab is needed and is being passed as a prop below. Not sure why VSCode is saying it can be removed.
+  const router = useRouter()
+  let defaultTab = router.query.tab ? String(router.query.tab) : 'Home'
+  defaultTab = defaultTab
```
Comment (mantikoros) [inline web/components/news-topics-tabs.tsx:45]: Reassigning variables is bad stylistically if you can avoid it

File: web/components/news-topics-tabs.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <Col className="mx-auto w-full gap-2 pb-8 sm:gap-6 sm:px-2 lg:pr-4">
+      <Row className="mx-2 items-center justify-between gap-4 sm:mx-0">
+        <h1
+          className={clsx(
+            'text-primary-700 !my-0 mb-4 inline-block text-2xl  font-normal sm:mb-6 sm:block sm:text-3xl'
+          )}
+        >
+          {defaultTab}
```
Comment (mantikoros) [inline web/components/news-topics-tabs.tsx:73]: shouldn't this be selectedTab?

File: web/components/news-topics-tabs.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+        onClick: () => {
+          track('news topic clicked', { tab: tab.title })
+        },
+      }))}
+    />
+  )
+}
+
+export function NewsTopicsTabs({ homeContent }: NewsTopicsTabsProps) {
+  const [selectedTab, setSelectedTab] = useState<string>('Home') // This selectedTab is needed and is being passed as a prop below. Not sure why VSCode is saying it can be removed.
```
Comment (mantikoros) [inline web/components/news-topics-tabs.tsx:42]: shouldn't this be useState(defaultTab) ?

File: web/components/news-topics-tabs.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        ...tab,
+        onClick: () => {
+          track('news topic clicked', { tab: tab.title })
+        },
+      }))}
+    />
+  )
+}
+
+export function NewsTopicsTabs({ homeContent }: NewsTopicsTabsProps) {
```
[the remark this answers is not in the document]
Comment (mantikoros) [reply web/components/news-topics-tabs.tsx:41]: actually, nvm. "homeContent" is more explicit

File: web/components/news-topics-tabs.tsx [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    { title: 'draft', content: <p>yay</p> },
+    { title: 'Reddit Blackout', content: <RedditBlackoutData /> },
+    { title: 'Draft3', content: <p>aalots of markets about your</p> },
+    { title: 'Draft4', content: <p>aalots of markets about </p> },
+  ]
+
+  return (
+    <Col className="mx-auto w-full gap-2 pb-8 sm:gap-6 sm:px-2 lg:pr-4">
+      <Row className="mx-2 items-center justify-between gap-4 sm:mx-0">
+        <h1
```
Comment (mantikoros) [inline web/components/news-topics-tabs.tsx:68]: The title may be unnecessary; the tab already has a title label in it.

**Review by mantikoros — COMMENTED** [review]
Very nice! Let me know when you've finished adding all the content.

**Comment by SirSaltyy** [comment]
@mantikoros this is ready to be merged to live I believe.

**Few outstanding things:**

-When clicking between tabs, there is horizontal shift due to some tabs having a scroll bar and some not.

-home/index.tsx has a ton of unused code sitting on it (eg. MainContent and YourDailyUpdates). I didn't want to delete it because it seemed like main also still had it just there? But I'm not sure since I'm still getting used to the conflict merger interface. So I'm guessing Ian intentionally left it there in case we wanted to switch back?

-Vercel prod and dev seemed to have failed deployment lol. But in my localhost everything works fine and none of the files have errors...

File: web/components/news-article-original.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        src={urlToImage}
+        alt={title}
+        height={200}
+      />
+
+      <a
+        className={'absolute inset-0 z-10'}
+        href={url}
+        target="_blank"
+        onClick={() => track('click news article', { article: url })}
```
Comment (mantikoros) [inline web/components/news-article-original.tsx:38]: the article title is the most important thing to track imho

File: web/pages/news-topics-data.tsx [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+          author="NBC News"
+          published_time={Date.UTC(2023, 5, 20, 12)}
+        />
+        <Container
+          header="New Markets"
+          containerContent={
+            <>
+              {contracts &&
+                newMarkets.length > 0 &&
+                newMarkets.map((contract) => (
```
Comment (mantikoros) [inline web/pages/news-topics-data.tsx:44]: the checks are unnecessary. {newMarkets.map( ...)} works fine.

File: web/components/widgets/container.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
+import { ReactNode } from 'react'
+import { Subtitle } from 'web/components/widgets/subtitle'
+import clsx from 'clsx'
+import { Col } from '../layout/col'
+
+export const Container = (props: {
```
Comment (mantikoros) [inline web/components/widgets/container.tsx:6]: "Container" is not a good name. What kind of container is it?

**Review by mantikoros — APPROVED** [review]
Ready to merge! Left a few comments on minor stylistic issues.

**Pull request merged**